mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1316,3 +1316,65 @@ def get_original(self):
             :return: The original data object.
             """
             return self._data
+
+
+class TextGenerationEvaluator(ModelEvaluator):
+    def can_evaluate(self, *, model_type, evaluator_config, **kwargs):
+        return True
+
+    def evaluate(
+        self,
+        *,
+        model,
+        model_type,
+        dataset,
+        run_id,
+        evaluator_config,
+        custom_metrics=None,
+        custom_artifacts=None,
+        baseline_model=None,
+        **kwargs,
+    ):
+        import evaluate
+
+        t = evaluate.load("toxicity")
+        df = dataset.features_data
+        preds = model.predict(df)
+        # How to log non-scalar metrics?
+        metrics = t.compute(predictions=preds)  # e.g. {'toxicity': [0.1, 0.2]}
+        print(preds, metrics)
+        # Should llm_predictions.csv contain metrics as well?
+        mlflow.llm.log_predictions(
+            inputs=df.to_dict("records"),
+            outputs=preds,
+            # How to get the prompts?
+            prompts=preds,
```

Comment:
```suggestion
            prompts=preds,
            metrics=list[{"toxicity": 0.1}, {"toxicity": 0.2}]
```

Reply:
outputs is list[dict[str, Any]]?

{"output_text": "...", "metric1": ..., metric2: "..."}

Reply:
mlflow.llm.log_predictions(
            inputs=df.to_dict("records"),
            outputs=preds,
            metadata=[{...}, {...}],  # -> ask Sunish